src/interface/sparse_pack_base.cpp

```diff
@@ -263,12 +323,12 @@ template SparsePackBase &SparsePackCache::BuildAndAdd<MeshBlockData<Real>>(
 
 std::string SparsePackCache::GetIdentifier(const PackDescriptor &desc) const {
   std::string identifier("");
-  for (const auto &flag : desc.flags)
-    identifier += flag.Name();
-  identifier += "____";
-  for (int i = 0; i < desc.vars.size(); ++i)
-    identifier += desc.vars[i] + std::to_string(desc.use_regex[i]);
-  identifier += "____";
+  for (const auto &vgroup : desc.var_groups) {
+    for (const auto &[vid, uid] : vgroup) {
+      identifier += std::to_string(uid) + "_";
+    }
+    identifier += "|";
+  }
```

Comment:
It probably matters not at all, since a string is just a vec of chars, but if the identifier is now just a string of UIDs, perhaps there's some compression/performance to be gained here by using a `std::vector<std::size_t>` instead of a s string?

(Actually it could also be the string is more performant, since a char is 1 byte vs 8 bytes for a long int. I guess do we expect UIDs to be 4 digits or longer?)

Reply:
Yeah, I thought a little bit about how to do that and it seemed like more complication than it was worth (since you can't just flatten the array of variable groups). And, as you point out, it is not clear that it will be faster anyway.